tests/lint/nextflow_config.py

```diff
@@ -100,6 +100,36 @@ def test_default_values_fail(self):
     )
 
 
+def test_catch_params_assignment_in_main_nf(self):
+    """Test linting fails if main.nf contains an assignment to a parameter from nextflow_schema.json."""
+    new_pipeline = self._make_pipeline_copy()
+    # Change reference of params.max_time to assignment
+    main_nf_file = Path(new_pipeline) / "main.nf"
+    with open(main_nf_file) as f:
+        content = f.read()
+        fail_content = re.sub(r"params.max_time ==", "params.max_time =", content)
```

Comment:
Here instead of replacing the added line in the original template, you can directly add a line anywhere in `main.nf`